### PR TITLE
fix: bqcc aws identity output-only and required in update

### PR DESCRIPTION
**Pull request by yuwenma**
### Change description

Fixes #

### Tests you have done

- [ ] Run `make ready-pr` to ensure this PR is ready for review.
- [ ] Perform necessary E2E testing for changed resources.

### CODE REVIEW

File: pkg/controller/direct/bigqueryconnection/connection_controller.go [hunk 1]
```diff
@@ -242,8 +254,9 @@ func (a *Adapter) Update(ctx context.Context, updateOp *directbase.UpdateOperati
 		return err
 	}
 	mapCtx := &direct.MapContext{}
-	desired := a.desired.DeepCopy()
-	connection := BigQueryConnectionConnectionSpec_ToProto(mapCtx, &desired.Spec)
+	connection := BigQueryConnectionConnectionSpec_ToProto(mapCtx, &a.desired.Spec)
+	connection = merge3(connection, a.desired.Status.ObservedState)
```
Comment (yuwenma) [inline pkg/controller/direct/bigqueryconnection/connection_controller.go:258]: This is a quick fix for GCP error "Error 400: Only 'aws.access_role.iam_role_id' field can be updated." where the `iam_role_id` is a output-only field, that the `CompareProtoMessage` considers the field needs to be changed.

I think the final solution should be a 3 way merge among the desired (`.spec`), previously applied (`.status.observedState`) and the actual.

File: pkg/controller/direct/bigqueryconnection/connection_controller.go [hunk 2]
```diff
@@ -232,6 +233,17 @@ func (a *Adapter) Create(ctx context.Context, createOp *directbase.CreateOperati
 	return setStatus(u, status)
 }
 
+// aws.accessRole.Identity is a output-only field in CREATE, it is required in the UPDATE.
+func merge3(desired *pb.Connection, previouslyApplied *krm.BigQueryConnectionConnectionObservedState) *pb.Connection {
+	if previouslyApplied != nil {
+		if previouslyApplied.Aws != nil {
```
Comment (jingyih) [inline pkg/controller/direct/bigqueryconnection/connection_controller.go:239]: nit: `if previouslyApplied != nil && previouslyApplied.Aws != nil {`

**Review by jingyih — COMMENTED** [review]
/lgtm

File: pkg/controller/direct/bigqueryconnection/connection_controller.go (hunk 2 above)
Comment (yuwenma) [reply pkg/controller/direct/bigqueryconnection/connection_controller.go:260]: Sure, done.
(I was thinking we may want to deal with other property merge, like Azure, SQL, Spanner, Spark, Salesforce)

**Review by jingyih — COMMENTED** [review]
/lgtm

File: pkg/controller/direct/bigqueryconnection/connection_controller.go [hunk 3]
```diff
@@ -21,6 +21,7 @@ import (
 
 	gcp "cloud.google.com/go/bigquery/connection/apiv1"
 	bigqueryconnectionpb "cloud.google.com/go/bigquery/connection/apiv1/connectionpb"
+	pb "cloud.google.com/go/bigquery/connection/apiv1/connectionpb"
```
Comment (cheftako) [inline pkg/controller/direct/bigqueryconnection/connection_controller.go:24]: This seems redundant with line 23. Why do we want 2 aliases for this package?

File: pkg/controller/direct/bigqueryconnection/connection_controller.go [hunk 4]
```diff
@@ -253,6 +254,14 @@ func (a *Adapter) Create(ctx context.Context, createOp *directbase.CreateOperati
 	return setStatus(u, status)
 }
 
+// aws.accessRole.Identity is a output-only field in CREATE, it is required in the UPDATE.
+func merge3(desired *pb.Connection, previouslyApplied *krm.BigQueryConnectionConnectionObservedState) *pb.Connection {
```
Comment (cheftako) [inline pkg/controller/direct/bigqueryconnection/connection_controller.go:258]: The function name seems confusing. Its not (currently) a 3 way merge. I would also expect something named  3 way merge to be a generic 3 way merge. This seems like a awsIdentityStateMerge() function. Can we rename it?

**Pull request merged**